### PR TITLE
[TEST] Checkout products on fixed promotion

### PR DESCRIPTION
**Pull request by szczecha**
I want to merge this change because it adds test for buying products on fixed promotion in checkout CORE_2102

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migrations are either absent or optimized for zero downtime
* [ ] The changes are covered by test cases

### CODE REVIEW

File: saleor/tests/e2e/checkout/utils/checkout_create.py [hunk 1]
```diff
@@ -45,6 +45,31 @@
...
+          }
+          tax {
+            amount
+          }
+        }
+        undiscountedTotalPrice {
+          amount
+        }
+        unitPrice {
+          net {
```
Comment (yellowee) [inline saleor/tests/e2e/checkout/utils/checkout_create.py:65]: @szczecha Shouldn't we return gross value here as well?

Comment (szczecha) [reply saleor/tests/e2e/checkout/utils/checkout_create.py]: thanks, fixed: https://github.com/saleor/saleor/pull/13649/commits/57c854d9aac92c419c262f145f67c0d3042b16d7

File: saleor/tests/e2e/checkout/test_checkout_products_on_fixed_promotion.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+from ..warehouse.utils import create_warehouse, update_warehouse
+from .utils import (
+    checkout_complete,
+    checkout_create,
+    checkout_delivery_method_update,
+    checkout_dummy_payment_create,
+)
+
+
+def prepare_product(
```
Comment (IKarbowiak) [inline saleor/tests/e2e/checkout/test_checkout_products_on_fixed_promotion.py:28]: I see that we are defining such method quite often in different test files, cannot we reuse it and just for example here extend it with the promotion data?

File: saleor/tests/e2e/checkout/test_checkout_products_on_fixed_promotion.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    checkout_dummy_payment_create(
+        e2e_not_logged_api_client, checkout_id, total_gross_amount
+    )
+
+    # Step 5 - Complete checkout.
+    order_data = checkout_complete(e2e_not_logged_api_client, checkout_id)
+
+    order_line = order_data["lines"][0]
+    assert order_data["status"] == "UNFULFILLED"
+    assert order_data["total"]["gross"]["amount"] == total_gross_amount
```
Comment (IKarbowiak) [inline saleor/tests/e2e/checkout/test_checkout_products_on_fixed_promotion.py:212]: Could we also check subtotal?

File: saleor/tests/e2e/checkout/test_checkout_products_on_fixed_promotion.py (hunk 2 above)
Comment (szczecha) [reply saleor/tests/e2e/checkout/test_checkout_products_on_fixed_promotion.py:28]: This is something we want to address in a separate PR

File: saleor/tests/e2e/checkout/test_checkout_products_on_fixed_promotion.py (hunk 3 above)
Comment (szczecha) [reply saleor/tests/e2e/checkout/test_checkout_products_on_fixed_promotion.py:213]: Added https://github.com/saleor/saleor/pull/13649/commits/2dbf9fb484ec55e7e5ff9588e57092c766b6b820

**Review by yellowee — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Pull request merged**